core/src/main/scala/cats/data/OptionT.scala

```diff
@@ -87,55 +107,212 @@ final case class OptionT[F[_], A](value: F[Option[A]]) {
     subflatMap(f)
 
+  /**
+   * Example:
+   * {{{
+   *  scala> import cats.implicits._
+   *  scala> import cats.data.OptionT
+   *
+   *  scala> val optionT: OptionT[List, Int] = OptionT[List, Int](List(Some(2), Some(3), Some(4)))
+   *  scala> optionT.flatMap(x => OptionT.when(x % 2 == 0)(x))
+   *  res0: OptionT[List, Int] = OptionT(List(Some(2), None, Some(4)))
+   * }}}
+   */
   def flatMap[B](f: A => OptionT[F, B])(implicit F: Monad[F]): OptionT[F, B] =
     flatMapF(a => f(a).value)
 
+  /**
+   * Example:
+   * {{{
+   *  scala> import cats.implicits._
+   *  scala> import cats.data.OptionT
+   *
+   *  scala> val optionT: OptionT[List, Int] = OptionT[List, Int](List(Some(2), Some(3), Some(4)))
+   *  scala> optionT.flatMapF(x => List(Option.when(x % 2 == 0)(x)))
+   *  res0: OptionT[List, Int] = OptionT(List(Some(2), None, Some(4)))
+   * }}}
+   */
   def flatMapF[B](f: A => F[Option[B]])(implicit F: Monad[F]): OptionT[F, B] =
     OptionT(F.flatMap(value)(_.fold(F.pure[Option[B]](None))(f)))
 
+  /**
+   * Example:
+   * {{{
+   *  scala> import cats.implicits._
+   *  scala> import cats.data.OptionT
+   *
+   *  scala> val optionT: OptionT[List, Int] = OptionT[List, Int](List(Some(2), Some(3), Some(4)))
+   *  scala> optionT.flatTransform(x => List(x.filter(_ % 2 == 0)))
+   *  res0: OptionT[List, Int] = OptionT(List(Some(2), None, Some(4)))
+   * }}}
+   */
   def flatTransform[B](f: Option[A] => F[Option[B]])(implicit F: Monad[F]): OptionT[F, B] =
     OptionT(F.flatMap(value)(f))
 
+  /**
+   * Example:
+   * {{{
+   *  scala> import cats.implicits._
+   *  scala> import cats.data.OptionT
+   *
+   *  scala> val optionT: OptionT[List, Int] = OptionT[List, Int](List(Some(2), Some(3), Some(4)))
+   *  scala> optionT.transform(_.filter(_ % 2 == 0))
+   *  res0: OptionT[List, Int] = OptionT(List(Some(2), None, Some(4)))
+   * }}}
+   */
   def transform[B](f: Option[A] => Option[B])(implicit F: Functor[F]): OptionT[F, B] =
     OptionT(F.map(value)(f))
 
+  /**
+   * Example:
+   * {{{
+   *  scala> import cats.implicits._
+   *  scala> import cats.data.OptionT
+   *
+   *  scala> val optionT: OptionT[List, Int] = OptionT[List, Int](List(Some(2), Some(3), Some(4)))
+   *  scala> optionT.subflatMap(x => Option.when(x % 2 == 0)(x))
+   *  res0: OptionT[List, Int] = OptionT(List(Some(2), None, Some(4)))
+   * }}}
+   */
   def subflatMap[B](f: A => Option[B])(implicit F: Functor[F]): OptionT[F, B] =
     transform(_.flatMap(f))
 
   /**
    * Perform an effect if the value inside the is a `None`, leaving the value untouched. Equivalent to [[orElseF]]
    * with an effect returning `None` as argument.
+   * 
+   * Example:
+   * {{{
+   *  scala> import cats.data.OptionT
+   *  scala> import scala.util.Try
+   *
+   *  scala> // prints "no value"
+   *  scala> val optionT: OptionT[Try, Int] = OptionT[Try, Int](Try(None))
+   *  scala> optionT.flatTapNone(Try(println("no value"))).value
```

Comment:
I think this example is also not really good for the same reason as one for `semiflatTap`.